app/src/ui/notifications/pull-request-checks-failed.tsx

```diff
@@ -0,0 +1,301 @@
+import * as React from 'react'
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { Row } from '../lib/row'
+import { OkCancelButtonGroup } from '../dialog/ok-cancel-button-group'
+import { PullRequest } from '../../models/pull-request'
+import { Dispatcher } from '../dispatcher'
+import { CICheckRunList } from '../check-runs/ci-check-run-list'
+import {
+  IRefCheck,
+  getLatestPRWorkflowRunsLogsForCheckRun,
+  getCheckRunActionsJobsAndLogURLS,
+  isFailure,
+} from '../../lib/ci-checks/ci-checks'
+import { CICheckRunLogs } from '../check-runs/ci-check-run-item-logs'
+import { Account } from '../../models/account'
+import { API, getHTMLURL } from '../../lib/api'
+import { Octicon, syncClockwise } from '../octicons'
+import * as OcticonSymbol from '../octicons/octicons.generated'
+import { Button } from '../lib/button'
+import { RepositoryWithGitHubRepository } from '../../models/repository'
+
+interface IPullRequestChecksFailedProps {
+  readonly dispatcher: Dispatcher
+  readonly shouldChangeRepository: boolean
+  readonly accounts: ReadonlyArray<Account>
+  readonly repository: RepositoryWithGitHubRepository
+  readonly pullRequest: PullRequest
+  readonly commitMessage: string
+  readonly commitSha: string
+  readonly checks: ReadonlyArray<IRefCheck>
+  readonly onSubmit: () => void
+  readonly onDismissed: () => void
+}
+
+interface IPullRequestChecksFailedState {
+  readonly switchingToPullRequest: boolean
+  readonly selectedCheckID: number
+  readonly checks: ReadonlyArray<IRefCheck>
+  readonly loadingActionWorkflows: boolean
+  readonly loadingActionLogs: boolean
+}
+
+/**
+ * Dialog to show the result of a CI check run.
+ */
+export class PullRequestChecksFailed extends React.Component<
+  IPullRequestChecksFailedProps,
+  IPullRequestChecksFailedState
+> {
+  private checkRunsLoadCancelled: boolean = false
+
+  public constructor(props: IPullRequestChecksFailedProps) {
+    super(props)
+
+    const { checks } = this.props
+
+    const selectedCheck = checks.find(check => isFailure(check)) ?? checks[0]
```

Comment:
```suggestion
    const selectedCheck = checks.find(isFailure) ?? checks[0]
```
Nit: Just a short hand that can make it slightly more readable. :P

Reply:
Oops! This was `!isSuccess` before, when I made the change to `isFailure` I didn't realize about this 😂  Thanks!